weaknesses/MASVS-CRYPTO/MASWE-0021.md

```diff
@@ -12,14 +12,41 @@ mappings:
 refs:
 - https://developer.android.com/privacy-and-security/cryptography#deprecated-functionality
 - https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-131Ar2.pdf
+- https://cwe.mitre.org/data/definitions/328.html
 - https://en.wikipedia.org/wiki/Collision_attack
 - https://csrc.nist.gov/pubs/ir/8547/ipd
-draft:
-  description: Utilizing broken hashing algorithms such as MD5 and SHA1 in a security
-    sensitive context may compromise data integrity and authenticity.
-  topics:
-  - Broken hashing algorithms (e.g. MD5, SHA1, etc.)
-status: placeholder
+status: new
 
 ---
 
+## Overview
+
+---
+
+Choosing a deprecated, risky or broken hash algorithm, that is insufficiently collision resistant, may compromise the integrity and authenticity of data at rest and in transit by making the app vulnerable to collision attacks.
```

Comment:
I have a friendly suggestion for narrowing the focus of this file:

```suggestion
Choosing a deprecated, risky, or broken hash algorithm that is insufficiently collision-resistant may compromise data integrity.
```

It's important to note that while hashes alone don't guarantee data authenticity. That's where MACs, like HMAC, come into play. I think it would be great to include HMAC (and the topic of data authenticity)  in MASWE-0024!